packages/round-manager/src/features/api/application.ts

```diff
@@ -73,6 +72,79 @@ export const getApplicationById = async (
   }
 };
 
+
+export const getApplicationsByRoundId = async (
+  roundId: string,
+  signerOrProvider: any
+): Promise<GrantApplication[]> => {
+
+  try {
+    // fetch chain id
+    const { chainId } = await signerOrProvider.getNetwork();
+
+    // query the subgraph for all rounds by the given account in the given program
+    const res = await graphql_fetch(
+      `
+        query GetGrantApplicationsByRoundId($roundId: String!, $status: String) {
+          roundProjects(where: {
+            round: $roundId
+      ` +
+        // TODO : uncomment when indexing IPFS via graph
+        // (status ? `status: $status` : ``)
+        // +
+      `
+          }) {
+            id
+            metaPtr {
+              protocol
+              pointer
+            }
+            status
+            round {
+              projectsMetaPtr {
+                protocol
+                pointer
+              }
+            }
+          }
+        }
+      `,
+      chainId,
+      { roundId }
+    );
+
+    const grantApplications: GrantApplication[] = [];
+
+    for (const project of res.data.roundProjects) {
+      const metadata = await fetchFromIPFS(project.metaPtr.pointer);
+
+      // const signature = metadata?.signature;
+      const application = metadata.application ? metadata.application : metadata;
+
+      grantApplications.push({
+        ...application,
+        status: project.status,
+        id: project.id,
+        projectsMetaPtr: project.round.projectsMetaPtr,
+      });
+
+    }
+
+    const grantApplicationsFromContract =
+      res.data.roundProjects.length > 0
+        ? await updateApplicationStatusFromContract(
+            grantApplications,
+            res.data.roundProjects[0].round.updateProjectsMetaPtr
```

Comment:
should this be ?
```suggestion
            project.round.projectsMetaPtr
```

Reply:
Oh, this is outside the for loop ! So project would be undefined here 😓